src/main/java/com/iprd/fhir/utils/Utils.java

```diff
@@ -483,6 +485,50 @@ public static byte[] zipStringData(String s) {
 			return null;
 		}
 	}
+	public static StringBuilder formatTimeDifferenceToHumanReadableString(long days, long hours, long minutes, long seconds) {
+		StringBuilder lastSync = new StringBuilder();
+		if (days > 0) {
+			lastSync.append(days).append(" day").append(days > 1 ? "s" : "");
+			if(hours > 0){
+				lastSync.append(", ").append(hours).append(" hour").append(hours > 1 ? "s" : "");
+			}
+		}
+		else if (hours > 0) {
+			lastSync.append(hours).append(" hour").append(hours > 1 ? "s" : "");
+			if(minutes > 0){
+				lastSync.append(", ").append(minutes).append(" minute").append(minutes > 1 ? "s" : "");
+			}
+		} else if (minutes > 0) {
+			lastSync.append(minutes).append(" minute").append(minutes > 1 ? "s" : "");
+			if(seconds > 0){
+				lastSync.append(", ").append(seconds).append(" second").append(seconds > 1 ? "s" : "");
+			}
+		} else {
+			if(seconds > 0){
+				lastSync.append(seconds).append(" second").append(seconds > 1 ? "s" : "");
+			}
+		}
+		return lastSync.append(" ago");
```

Comment:
.toString() can be done here itself by returning String instead of StringBuilder.

Reply:
In terms of efficiency, returning a StringBuilder and then converting it to a string when needed using .toString() is generally more efficient than converting to a string and returning it directly.